src/aktualizr_lite/helpers.cc

```diff
@@ -88,21 +106,22 @@ LiteClient::LiteClient(Config &config_in) : config(std::move(config_in)), primar
   headers.push_back(header);
   add_apps_header(headers, config.pacman);
 
-  std::pair<bool, Uptane::Target> pair = finalizeIfNeeded(*storage, config.pacman);
-  headers.emplace_back("x-ats-target: " + pair.second.filename());
-  headers.emplace_back("x-ats-tags: " + boost::algorithm::join(config.pacman.tags, ","));
-
   http_client = std::make_shared<HttpClient>(&headers);
   report_queue = std_::make_unique<ReportQueue>(config, http_client);
+  package_manager = PackageManagerFactory::makePackageManager(config.pacman, config.bootloader, storage, http_client);
 
-  if (pair.first) {
-    notifyInstallFinished(pair.second, data::ResultCode::Numeric::kOk);
-  }
+  std::pair<Uptane::Target, data::ResultCode::Numeric> pair = finalizeIfNeeded(*package_manager, *storage, config.pacman);
+  http_client->updateHeader("x-ats-target", pair.first.filename());
+  http_client->updateHeader("x-ats-tags", boost::algorithm::join(config.pacman.tags, ","));
```

Comment:
It looks like [`updateHeader`](https://github.com/advancedtelematic/aktualizr/blob/master/src/libaktualizr/http/httpclient.cc#L261) requires the header to already be defined. The "x-ats-tags" header can stay where it was, so the real problem you have is with the "x-ats-target" header. I can see two ways to deal with it:
* Define it above with an value like "?" and then updateHeader will work
* The calls you make to package-manager don't require a real HttpClient, so you could just pass a default constructed HttpClient just to that object

Reply:
Hm, looking at the code it seems http is not really used by any package manager, but I guess it could at some point?

And nice catch, I had the impression it would update or add to the list of known headers but you are right :-)

Guess I will go with the first suggestion to be more aligned with what package-manager expects.

Reply:
Pushed an update following your first suggestion, setting to 'unknown' first and later updating based on the output of finalizeIfNeeded.